catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SolrClientAdaptor.java

```diff
@@ -0,0 +1,186 @@
+
+    private Function<SolrClient, CacheSolrMetacardClient> metacardClientSupplierFunction;
+
+    private Function<SolrClient, InitializedSolrClientAdaptor> clientAdaptorSupplierFunction;
+
+    /**
+     * Interface implemented by the different states the {@link SolrClientAdaptor} can be in.
+     * All operations on the {@link SolrClientAdaptor} will be delegate to the current
+     * state implementation.
+     */
+    interface State {
+        /**
+         * Commits Solr transactions.
+         *
+         * @see SolrClient#commit()
+         */
+        void commit() throws SolrServerException, IOException;
+
+        /**
+         * Closes the Solr connection.
+         *
+         * @see SolrClient#close()
+         */
+        void close() throws IOException;
+
+        /**
+         * Deletes Solr documents that match the query provided.
+         *
+         * @see SolrClient#deleteByQuery(String)
+         */
+        UpdateResponse deleteByQuery(String query) throws SolrServerException, IOException;
+    }
+
+    private final String coreName;
+
+    private final FilterAdapter filterAdapter;
+
+    private final SolrClientFactory solrClientFactory;
+
+    private final SolrFilterDelegateFactory solrFilterDelegateFactory;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrClientAdaptor.class);
+
+    private Future<SolrClient> solrClientFuture;
+
+    private volatile SolrMetacardClient client = NoOpSolrMetacardClient.getInstance();
+
+    private volatile State state = UninitializedSolrClientAdaptor.getInstance();
+
+    SolrClientAdaptor(String coreName, FilterAdapter filterAdapter,
+            SolrClientFactory solrClientFactory,
+            SolrFilterDelegateFactory solrFilterDelegateFactory) {
+        this.metacardClientSupplierFunction =
+                (solrClient) -> new CacheSolrMetacardClient(solrClient,
+                        filterAdapter,
+                        solrFilterDelegateFactory);
+        this.clientAdaptorSupplierFunction = (solrClient) -> new InitializedSolrClientAdaptor(
+                solrClient);
+        this.coreName = coreName;
+        this.filterAdapter = filterAdapter;
+        this.solrClientFactory = solrClientFactory;
+        this.solrFilterDelegateFactory = solrFilterDelegateFactory;
+    }
+
+    public void init() {
+        RetryPolicy retryPolicy = new RetryPolicy();
+
+        Failsafe.with(retryPolicy)
+                .onRetry((exception) -> LOGGER.debug(
+                        "Failed to get Solr client for SolrCache. Retrying...",
+                        exception))
+                .onSuccess((result, context) -> LOGGER.debug(
+                        "SolrCache successfully initialized after {} retries",
+                        context.getExecutions()))
+                .run(this::getSolrClient);
+    }
+
+    //For unit testing purposes.
+    State getState() {
+        return state;
+    }
+
+    //For unit testing purposes.
```

Comment:
@lessarderic and I discussed quite a bit. We landed on this option due to the limitations around mocking creating a new `CacheSolrMetacardClient` and `InitializedSolrClientAdaptor`. We were intentionally avoiding PowerMock.